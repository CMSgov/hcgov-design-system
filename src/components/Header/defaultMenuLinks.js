import localeLink from './localeLink';
import loginLink from './loginLink';
import { translate } from '../i18n';

/**
 * Default menu links for each header variation.
 * Apps can import this method into their app if they need to
 * extend the existing default list of menu links.
 * @param {Function} translate - i18n translator
 * @param {String} locale
 * @param {Boolean} deConsumer
 * @param {String} subpath
 * @param {String} primaryDomain
 * @param {String} switchLocaleLink
 * @param {boolean} hideLoginLink
 * @param {boolean} hideLanguageSwitch
 * @returns {Object}
 */
export function defaultMenuLinks(
  locale = 'en',
  deConsumer,
  subpath,
  primaryDomain = '',
  switchLocaleLink,
  hideLoginLink,
  hideLanguageSwitch
) {
  const isSpanish = locale === 'es';
  const ffmLocalePath = isSpanish ? 'es_MX' : 'en_US';
  // We import and set i18n options within this method, for
  // scenarios where an app needs to import this method and
  // extend the existing list of default links
  const t = translate;
  const i18nOptions = { lng: locale };

  const minimal = [];
  if (!hideLanguageSwitch) minimal.push(localeLink(t, locale, subpath, switchLocaleLink));
  if (!hideLoginLink) minimal.push(loginLink(t, deConsumer, primaryDomain));

  return {
    'logged-out': minimal,
    'logged-in': [
      {
        label: t('header.myApplicationsAndCoverage', i18nOptions),
        href: `${primaryDomain}/marketplace/auth/global/${ffmLocalePath}/myProfile#landingPage`,
      },
      {
        label: t('header.myProfile', i18nOptions),
        href: `${primaryDomain}/marketplace/auth/global/${ffmLocalePath}/myProfile#settings`,
      },
    ],
  };
}

export default defaultMenuLinks;
