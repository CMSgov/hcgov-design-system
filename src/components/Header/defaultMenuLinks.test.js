import defaultMenuLinks, {
  defaultMenuLinks as namedExportDefaultMenuLinks,
} from './defaultMenuLinks';

describe('MenuList', function () {
  it('includes a named export', () => {
    // Some apps may need to import the default menu links in order to extend them
    expect(typeof namedExportDefaultMenuLinks).toBe('function');
  });

  describe('English', () => {
    it('returns array of menu list objects', () => {
      expect(defaultMenuLinks()).toMatchSnapshot();
    });

    it('returns array of menu list objects with subpath', () => {
      expect(defaultMenuLinks('en', false, 'tax-tool/')).toMatchSnapshot();
    });

    it('returns array of menu list objects with absolute URLs', () => {
      expect(
        defaultMenuLinks('en', false, undefined, 'https://www.healthcare.gov')
      ).toMatchSnapshot();
    });

    it('returns array of menu list objects with custom locale switch link', () => {
      expect(
        defaultMenuLinks(
          'en',
          false,
          undefined,
          undefined,
          'https://ayudalocal.cuidadodesalud.gov/es'
        )
      ).toMatchSnapshot();
    });
  });

  describe('Spanish', () => {
    it('returns array of menu list objects', () => {
      expect(defaultMenuLinks('es')).toMatchSnapshot();
    });

    it('returns array of menu list objects with subpath', () => {
      expect(defaultMenuLinks('es', false, 'tax-tool/')).toMatchSnapshot();
    });

    it('returns array of menu list objects with absolute URLs', () => {
      expect(
        defaultMenuLinks('es', false, undefined, 'https://www.healthcare.gov')
      ).toMatchSnapshot();
    });

    it('returns array of menu list objects with custom locale switch link', () => {
      expect(
        defaultMenuLinks(
          'es',
          false,
          undefined,
          undefined,
          'https://localhelp.healthcare.gov'
        )
      ).toMatchSnapshot();
    });
  });
});
