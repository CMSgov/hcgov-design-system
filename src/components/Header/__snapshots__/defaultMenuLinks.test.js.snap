// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuList English returns array of menu list objects 1`] = `
Object {
  "home": Array [
    Object {
      "href": "#TODO",
      "label": "TODO: defaultMenuLinks.js",
    },
  ],
  "logged-in": Array [
    Object {
      "href": "/marketplace/auth/global/en_US/myProfile#landingPage",
      "label": "header.myApplicationsAndCoverage | {\\"lng\\":\\"en\\"}",
    },
    Object {
      "href": "/marketplace/auth/global/en_US/myProfile#settings",
      "label": "header.myProfile | {\\"lng\\":\\"en\\"}",
    },
    Object {
      "href": "/logout",
      "label": "header.logout | {\\"lng\\":\\"en\\"}",
    },
  ],
  "minimal": Array [
    Object {
      "href": "/login",
      "label": "header.login",
    },
    Object {
      "href": "https://www.cuidadodesalud.gov/es/",
      "label": "header.espa単ol",
    },
  ],
}
`;

exports[`MenuList English returns array of menu list objects with absolute URLs 1`] = `
Object {
  "home": Array [
    Object {
      "href": "#TODO",
      "label": "TODO: defaultMenuLinks.js",
    },
  ],
  "logged-in": Array [
    Object {
      "href": "https://www.healthcare.gov/marketplace/auth/global/en_US/myProfile#landingPage",
      "label": "header.myApplicationsAndCoverage | {\\"lng\\":\\"en\\"}",
    },
    Object {
      "href": "https://www.healthcare.gov/marketplace/auth/global/en_US/myProfile#settings",
      "label": "header.myProfile | {\\"lng\\":\\"en\\"}",
    },
    Object {
      "href": "https://www.healthcare.gov/logout",
      "label": "header.logout | {\\"lng\\":\\"en\\"}",
    },
  ],
  "minimal": Array [
    Object {
      "href": "https://www.healthcare.gov/login",
      "label": "header.login",
    },
    Object {
      "href": "https://www.cuidadodesalud.gov/es/",
      "label": "header.espa単ol",
    },
  ],
}
`;

exports[`MenuList English returns array of menu list objects with custom locale switch link 1`] = `
Object {
  "home": Array [
    Object {
      "href": "#TODO",
      "label": "TODO: defaultMenuLinks.js",
    },
  ],
  "logged-in": Array [
    Object {
      "href": "/marketplace/auth/global/en_US/myProfile#landingPage",
      "label": "header.myApplicationsAndCoverage | {\\"lng\\":\\"en\\"}",
    },
    Object {
      "href": "/marketplace/auth/global/en_US/myProfile#settings",
      "label": "header.myProfile | {\\"lng\\":\\"en\\"}",
    },
    Object {
      "href": "/logout",
      "label": "header.logout | {\\"lng\\":\\"en\\"}",
    },
  ],
  "minimal": Array [
    Object {
      "href": "/login",
      "label": "header.login",
    },
    Object {
      "href": "https://ayudalocal.cuidadodesalud.gov/es",
      "label": "header.espa単ol",
    },
  ],
}
`;

exports[`MenuList English returns array of menu list objects with subpath 1`] = `
Object {
  "home": Array [
    Object {
      "href": "#TODO",
      "label": "TODO: defaultMenuLinks.js",
    },
  ],
  "logged-in": Array [
    Object {
      "href": "/marketplace/auth/global/en_US/myProfile#landingPage",
      "label": "header.myApplicationsAndCoverage | {\\"lng\\":\\"en\\"}",
    },
    Object {
      "href": "/marketplace/auth/global/en_US/myProfile#settings",
      "label": "header.myProfile | {\\"lng\\":\\"en\\"}",
    },
    Object {
      "href": "/logout",
      "label": "header.logout | {\\"lng\\":\\"en\\"}",
    },
  ],
  "minimal": Array [
    Object {
      "href": "/login",
      "label": "header.login",
    },
    Object {
      "href": "https://www.cuidadodesalud.gov/es/tax-tool/",
      "label": "header.espa単ol",
    },
  ],
}
`;

exports[`MenuList Spanish returns array of menu list objects 1`] = `
Object {
  "home": Array [
    Object {
      "href": "#TODO",
      "label": "TODO: defaultMenuLinks.js",
    },
  ],
  "logged-in": Array [
    Object {
      "href": "/marketplace/auth/global/es_MX/myProfile#landingPage",
      "label": "header.myApplicationsAndCoverage | {\\"lng\\":\\"es\\"}",
    },
    Object {
      "href": "/marketplace/auth/global/es_MX/myProfile#settings",
      "label": "header.myProfile | {\\"lng\\":\\"es\\"}",
    },
    Object {
      "href": "/logout",
      "label": "header.logout | {\\"lng\\":\\"es\\"}",
    },
  ],
  "minimal": Array [
    Object {
      "href": "/login",
      "label": "header.login",
    },
    Object {
      "href": "https://www.healthcare.gov/",
      "label": "header.english",
    },
  ],
}
`;

exports[`MenuList Spanish returns array of menu list objects with absolute URLs 1`] = `
Object {
  "home": Array [
    Object {
      "href": "#TODO",
      "label": "TODO: defaultMenuLinks.js",
    },
  ],
  "logged-in": Array [
    Object {
      "href": "https://www.healthcare.gov/marketplace/auth/global/es_MX/myProfile#landingPage",
      "label": "header.myApplicationsAndCoverage | {\\"lng\\":\\"es\\"}",
    },
    Object {
      "href": "https://www.healthcare.gov/marketplace/auth/global/es_MX/myProfile#settings",
      "label": "header.myProfile | {\\"lng\\":\\"es\\"}",
    },
    Object {
      "href": "https://www.healthcare.gov/logout",
      "label": "header.logout | {\\"lng\\":\\"es\\"}",
    },
  ],
  "minimal": Array [
    Object {
      "href": "https://www.healthcare.gov/login",
      "label": "header.login",
    },
    Object {
      "href": "https://www.healthcare.gov/",
      "label": "header.english",
    },
  ],
}
`;

exports[`MenuList Spanish returns array of menu list objects with custom locale switch link 1`] = `
Object {
  "home": Array [
    Object {
      "href": "#TODO",
      "label": "TODO: defaultMenuLinks.js",
    },
  ],
  "logged-in": Array [
    Object {
      "href": "/marketplace/auth/global/es_MX/myProfile#landingPage",
      "label": "header.myApplicationsAndCoverage | {\\"lng\\":\\"es\\"}",
    },
    Object {
      "href": "/marketplace/auth/global/es_MX/myProfile#settings",
      "label": "header.myProfile | {\\"lng\\":\\"es\\"}",
    },
    Object {
      "href": "/logout",
      "label": "header.logout | {\\"lng\\":\\"es\\"}",
    },
  ],
  "minimal": Array [
    Object {
      "href": "/login",
      "label": "header.login",
    },
    Object {
      "href": "https://localhelp.healthcare.gov",
      "label": "header.english",
    },
  ],
}
`;

exports[`MenuList Spanish returns array of menu list objects with subpath 1`] = `
Object {
  "home": Array [
    Object {
      "href": "#TODO",
      "label": "TODO: defaultMenuLinks.js",
    },
  ],
  "logged-in": Array [
    Object {
      "href": "/marketplace/auth/global/es_MX/myProfile#landingPage",
      "label": "header.myApplicationsAndCoverage | {\\"lng\\":\\"es\\"}",
    },
    Object {
      "href": "/marketplace/auth/global/es_MX/myProfile#settings",
      "label": "header.myProfile | {\\"lng\\":\\"es\\"}",
    },
    Object {
      "href": "/logout",
      "label": "header.logout | {\\"lng\\":\\"es\\"}",
    },
  ],
  "minimal": Array [
    Object {
      "href": "/login",
      "label": "header.login",
    },
    Object {
      "href": "https://www.healthcare.gov/tax-tool/",
      "label": "header.english",
    },
  ],
}
`;
